### PR TITLE
schedule: fix the issue that stale operator when scatting regions

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
Problem produce:
```
create table x(a int);
split table x between (0) and (1000000) regions 1000;
```
And found many stale operators:
```
[2019/11/04 23:43:33.176 +08:00] [INFO] [operator_controller.go:122] ["stale operator"] [region-id=9422] [takes=24.486979339s] [operator="\"scatter-region {scatter region: stores [1 6 7] to [4 6 7]} (kind:leader,region, region:9422(2019,8), createAt:2019-11-04 23:43:08.689465805 +0800 CST m=+242.334032834, startAt:2019-11-04 23:43:08.689582143 +0800 CST m=+242.334149184, currentStep:2, steps:[add learner peer 10733 on store 4, promote learner peer 10733 on store 4 to voter, transfer leader from store 6 to store 4, remove peer on store 1])\""] [diff=2] [latest="{\"id\":9422,\"start_key\":\"dIAAAAAAAAD/K19ygAAAAAD/HeRYAAAAAAD6\",\"end_key\":\"dIAAAAAAAAD/K19ygAAAAAD/HehAAAAAAAD6\",\"region_epoch\":{\"conf_ver\":10,\"version\":2019},\"peers\":[{\"id\":9423,\"store_id\":1},{\"id\":9424,\"store_id\":7},{\"id\":9425,\"store_id\":6},{\"id\":10733,\"store_id\":4}]}"] [origin="{\"conf_ver\":8,\"version\":2019}"]
[2019/11/04 23:43:33.176 +08:00] [INFO] [operator_controller.go:122] ["stale operator"] [region-id=8802] [takes=24.591804222s] [operator="\"scatter-region {scatter region: stores [1 6 7] to [4 6 7]} (kind:leader,region, region:8802(2019,8), createAt:2019-11-04 23:43:08.584864973 +0800 CST m=+242.229431969, startAt:2019-11-04 23:43:08.584949534 +0800 CST m=+242.229516563, currentStep:2, steps:[add learner peer 10547 on store 4, promote learner peer 10547 on store 4 to voter, transfer leader from store 6 to store 4, remove peer on store 1])\""] [diff=2] [latest="{\"id\":8802,\"start_key\":\"dIAAAAAAAAD/K19ygAAAAAD/G4bgAAAAAAD6\",\"end_key\":\"dIAAAAAAAAD/K19ygAAAAAD/G4rIAAAAAAD6\",\"region_epoch\":{\"conf_ver\":10,\"version\":2019},\"peers\":[{\"id\":8803,\"store_id\":1},{\"id\":8804,\"store_id\":7},{\"id\":8805,\"store_id\":6},{\"id\":10547,\"store_id\":4}]}"] [origin="{\"conf_ver\":8,\"version\":2019}"]
[2019/11/04 23:43:33.176 +08:00] [INFO] [operator_controller.go:122] ["stale operator"] [region-id=8634] [takes=24.622171763s] [operator="\"scatter-region {scatter region: stores [1 6 7] to [4 5 6]} (kind:region, region:8634(2019,8), createAt:2019-11-04 23:43:08.554697786 +0800 CST m=+242.199264820, startAt:2019-11-04 23:43:08.554785208 +0800 CST m=+242.199352229, currentStep:2, steps:[add learner peer 10496 on store 4, promote learner peer 10496 on store 4 to voter, remove peer on store 1, add learner peer 10497 on store 5, promote learner peer 10497 on store 5 to voter, remove peer on store 7])\""] [diff=2] [latest="{\"id\":8634,\"start_key\":\"dIAAAAAAAAD/K19ygAAAAAD/GuLQAAAAAAD6\",\"end_key\":\"dIAAAAAAAAD/K19ygAAAAAD/Gua4AAAAAAD6\",\"region_epoch\":{\"conf_ver\":10,\"version\":2019},\"peers\":[{\"id\":8635,\"store_id\":1},{\"id\":8636,\"store_id\":7},{\"id\":8637,\"store_id\":6},{\"id\":10496,\"store_id\":4}]}"] [origin="{\"conf_ver\":8,\"version\":2019}"]
[2019/11/04 23:43:33.177 +08:00] [INFO] [operator_controller.go:122] ["stale operator"] [region-id=9462] [takes=24.480428926s] [operator="\"scatter-region {scatter region: stores [1 6 7] to [4 6 7]} (kind:region, region:9462(2019,8), createAt:2019-11-04 23:43:08.696571071 +0800 CST m=+242.341138082, startAt:2019-11-04 23:43:08.696719225 +0800 CST m=+242.341286242, currentStep:2, steps:[add learner peer 10745 on store 4, promote learner peer 10745 on store 4 to voter, remove peer on store 1])\""] [diff=2] [latest="{\"id\":9462,\"start_key\":\"dIAAAAAAAAD/K19ygAAAAAD/HgtoAAAAAAD6\",\"end_key\":\"dIAAAAAAAAD/K19ygAAAAAD/Hg9QAAAAAAD6\",\"region_epoch\":{\"conf_ver\":10,\"version\":2019},\"peers\":[{\"id\":9463,\"store_id\":1},{\"id\":9464,\"store_id\":7},{\"id\":9465,\"store_id\":6},{\"id\":10745,\"store_id\":4}]}"] [origin="{\"conf_ver\":8,\"version\":2019}"]
```
Cause of the mismatch check of  the `AddLightLearner`. 
### What is changed and how it works?
consistent with `AddLearner`

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/schedule/operator/operator.go [hunk 1]
```diff
@@ -423,7 +423,7 @@ type AddLightLearner struct {
 
 // ConfVerChanged returns true if the conf version has been changed by this step
 func (al AddLightLearner) ConfVerChanged(region *core.RegionInfo) bool {
-	if p := region.GetStoreLearner(al.ToStore); p != nil {
+	if p := region.GetStorePeer(al.ToStore); p != nil {
```
Comment (lhy1024) [inline server/schedule/operator/operator.go]: It seemed to be opposite with  #1732

File: server/schedule/operator_controller_test.go [hunk 2]
```diff
@@ -339,83 +339,96 @@ func (t *testOperatorControllerSuite) TestDispatchUnfinishedStep(c *C) {
...
-	c.Assert(len(stream.MsgCh()), Equals, 4)
-	c.Assert(controller.GetOperator(1), IsNil)
+	for _, steps := range testSteps {
+		// Create an operator
+		op := operator.NewOperator("test", "test", 1, epoch,
+			operator.OpRegion, steps...)
+		c.Assert(controller.AddOperator(op), Equals, true)
+		c.Assert(len(stream.MsgCh()), Equals, 1)
+
+		// Create region2 witch is cloned from the original region.
```
Comment (rleungx) [inline server/schedule/operator_controller_test.go]:
```suggestion
		// Create region2 which is cloned from the original region.
```

File: server/schedule/operator_controller_test.go [hunk 3]
```diff
@@ -339,83 +339,96 @@ func (t *testOperatorControllerSuite) TestDispatchUnfinishedStep(c *C) {
...
+				{Id: 3, StoreId: 3, IsLearner: true},
+			}),
+			core.WithIncConfVer(),
+		)
+		c.Assert(region2.GetPendingPeers(), NotNil)
+		c.Assert(steps[0].IsFinish(region2), Equals, false)
+		controller.Dispatch(region2, DispatchFromHeartBeat)
+
+		// In this case, the conf version has been changed, but the
+		// peer added is in peeding state, the operator should not be
```
Comment (rleungx) [inline server/schedule/operator_controller_test.go]:
```suggestion
		// peer added is in pending state, the operator should not be
```

**Comment by nolouch** [comment]
PTAL @rleungx

**Pull request merged**